.github/workflows/test_tarball.yml

```
@@ -0,0 +1,216 @@
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+#
+# This github CI script installs t8code and runs its tests for various configurations.
+# We compile sc and p4est as thirdparty libraries and use caching to only trigger a
+# new installation of them when their versions have changed in t8code.
+#
+# Note: To manually enforce sc and p4est installation, either increase the counter
+#       in the "key:" entries of the sc and p4est steps or set the variables
+#       SC_IGNORE_CACHE and P4EST_IGNORE_CACHE to 1 in the respective steps.
+
+env:
+  MAKEFLAGS: "-j2 V=0"
+
+
+on:
+  push:
+    branches:
+      - main
+      - develop
+      - CI-*tarball* # for testing this script, all feature branches with "tarball" in their name
+  pull_request:
+    branches:
+      - main
+      - develop
+  workflow_dispatch: 
+
+jobs:
+  build:
+
+    if: (github.event_name == 'schedule' && github.repository == 'DLR-AMR/t8code') || (github.event_name != 'schedule')
+    runs-on: ubuntu-20.04
+    container: dlramr/t8code-ubuntu:t8-dependencies
+    timeout-minutes: 90
+    steps:
+#
+# Setup and bootstrap
+#
+    - uses: actions/checkout@v4
+      with: 
+        fetch-depth: 0
+    - name: Update packages
+      run: apt-get update && apt-get upgrade -y
+      # This seems to be necessary because of the docker container
+    - name: disable ownership checks
+      run: git config --global --add safe.directory '*'
+    - name: init submodules
+      run: git submodule init
+    - name: update submodules
+      run: git submodule update
+    - name: bootstrap
+      run: ./bootstrap
+#       SC installation
+#
+#
+#  T8CODE 
+#     with p4est and sc as internal dependencies which is needed for make dist
+#
+#
+    # build config vars
+    - name: build CFLAGS and CXXFLAGS variables
+      run: echo CFLAGS_var="-Wall -pedantic -O0" >> $GITHUB_ENV
+        && echo CXXFLAGS_var="-Wall -pedantic -O0" >> $GITHUB_ENV
+    # Note: We want to use '-Werror', but if we already provide it at the configure step
+    #       we get errors in configure that we cannot remove (for example the autotools way
+    #       of checking for libm results in a compiler warning).
+    #       Thus, we add '-Werror' later at the make step.
+    - name: less-test-option
+      if: github.event_name == 'pull_request'
+      run: export LESS_TEST_OPTION="--enable-less-tests"
+           && echo LESS_TEST_OPTION="$LESS_TEST_OPTION" >> $GITHUB_ENV
+    - name: build config variables
+      run: export CONFIG_OPTIONS="--without-blas ${LESS_TEST_OPTION}"
+           && export CONFIG_DEBUG="$CONFIG_OPTIONS --enable-debug --enable-mpi"
+           && echo CONFIG_OPTIONS="$CONFIG_OPTIONS" >> $GITHUB_ENV
+           && echo CONFIG_DEBUG="$CONFIG_DEBUG" >> $GITHUB_ENV
+    - name: Check vars
+      run: echo "[$CONFIG_DEBUG]"
+    - name: configure debug
+      run: mkdir build && cd build && ../configure $CONFIG_DEBUG CFLAGS="$CFLAGS_var" CXXFLAGS="$CXXFLAGS_var"
```

```suggestion
    - name: configure
      run: mkdir build && cd build && ../configure
```
Since the user specifies the build options in the end, we can run the configure script without any arguments